compiler crash during compilation with operation on type parametrized class inside quote.

## Compiler version

3.1.1-RC1
3.1.0

## Minimized code


```Scala
package x

import scala.quoted._

object Impl {

      sealed trait UpdateOp[+T]
      case class Assignment[T](value:Expr[T]) extends UpdateOp[T]
      case class Update(operation:Expr[Unit]) extends UpdateOp[Nothing]

      def genRead[B:Type](newBuilder: Expr[B],
                          readVal: (Expr[B]) => UpdateOp[B]
                          )(using Quotes): Expr[B] =
          '{
                var x = $newBuilder
                  ${readVal[B]('x) match {
                    case Assignment(value) => '{ x = $value }
                    case Update(operation) => operation
                  }}
                x
          }

}
```

## Output (click arrow to expand)
<details>

```scala
ssh@rssh4 crash-ts-ownity % sbt compile
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by sbt.TrapExit$ (file:/Users/rssh/.sbt/boot/scala-2.12.14/org.scala-sbt/sbt/1.5.7/run_2.12-1.5.7.jar)
WARNING: Please consider reporting this to the maintainers of sbt.TrapExit$
WARNING: System::setSecurityManager will be removed in a future release
[info] welcome to sbt 1.5.7 (Homebrew Java 17)
[info] loading global plugins from /Users/rssh/.sbt/1.0/plugins
[info] loading project definition from /Users/rssh/tests/dotty/crash-ts-ownity/project
[info] loading settings for project root from build.sbt ...
[info] set current project to test (in build file:/Users/rssh/tests/dotty/crash-ts-ownity/)
[info] Executing in batch mode. For better performance use sbt's shell
[info] compiling 1 Scala source to /Users/rssh/tests/dotty/crash-ts-ownity/target/scala-3.1.1-RC1/classes ...
[info] exception occurred while typechecking /Users/rssh/tests/dotty/crash-ts-ownity/src/main/scala/x/X.scala
[info] exception occurred while compiling /Users/rssh/tests/dotty/crash-ts-ownity/src/main/scala/x/X.scala
java.lang.AssertionError: assertion failed: TS[2, 1, 0] attempted to take ownership of T which is already owned by committable TS[99, 2, 1, 0] while compiling /Users/rssh/tests/dotty/crash-ts-ownity/src/main/scala/x/X.scala
[error] ## Exception when compiling 1 sources to /Users/rssh/tests/dotty/crash-ts-ownity/target/scala-3.1.1-RC1/classes
[error] java.lang.AssertionError: assertion failed: TS[2, 1, 0] attempted to take ownership of T which is already owned by committable TS[99, 2, 1, 0]
[error] scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[error] dotty.tools.dotc.core.TyperState.includeVar(TyperState.scala:240)
[error] dotty.tools.dotc.core.TyperState.$anonfun$2$$anonfun$1(TyperState.scala:204)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.immutable.List.foreach(List.scala:333)
[error] dotty.tools.dotc.core.TyperState.$anonfun$2(TyperState.scala:204)
[error] scala.collection.immutable.List.forall(List.scala:386)
[error] dotty.tools.dotc.core.TyperState.mergeConstraintWith(TyperState.scala:206)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArgs(ProtoTypes.scala:416)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.fold(ProtoTypes.scala:488)
[error] dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:5921)
[error] dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:5959)
[error] dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.traverse(Implicits.scala:594)
[error] dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] dotty.tools.dotc.typer.ProtoTypes$PolyProto.fold(ProtoTypes.scala:599)
[error] dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:5921)
[error] dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:5959)
[error] dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.traverse(Implicits.scala:594)
[error] dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] dotty.tools.dotc.typer.ProtoTypes$SelectionProto.fold(ProtoTypes.scala:214)
[error] dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:5921)
[error] dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:5959)
[error] dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.traverse(Implicits.scala:594)
[error] dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] dotty.tools.dotc.typer.ProtoTypes$ViewProto.fold(ProtoTypes.scala:545)
[error] dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:5921)
[error] dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:5959)
[error] dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.traverse(Implicits.scala:594)
[error] dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.apply(Implicits.scala:599)
[error] dotty.tools.dotc.typer.ImplicitRunInfo.recur$1(Implicits.scala:675)
[error] dotty.tools.dotc.typer.ImplicitRunInfo.computeIScope(Implicits.scala:686)
[error] dotty.tools.dotc.typer.ImplicitRunInfo.$anonfun$1(Implicits.scala:760)
[error] dotty.tools.dotc.util.ReadOnlyMap.getOrElse(ReadOnlyMap.scala:23)
[error] dotty.tools.dotc.typer.ImplicitRunInfo.implicitScope(Implicits.scala:760)
[error] dotty.tools.dotc.typer.ImplicitRunInfo.implicitScope$(Implicits.scala:543)
[error] dotty.tools.dotc.Run.implicitScope(Run.scala:40)
[error] dotty.tools.dotc.typer.Implicits$ImplicitSearch.implicitScope(Implicits.scala:1454)
[error] dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicit(Implicits.scala:1417)
[error] dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicit(Implicits.scala:1426)
[error] dotty.tools.dotc.typer.Implicits$ImplicitSearch.bestImplicit(Implicits.scala:1451)
[error] dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:991)
[error] dotty.tools.dotc.typer.Implicits.inferImplicit$(Implicits.scala:771)
[error] dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:107)
[error] dotty.tools.dotc.typer.Implicits.inferView(Implicits.scala:817)
[error] dotty.tools.dotc.typer.Implicits.inferView$(Implicits.scala:771)
[error] dotty.tools.dotc.typer.Typer.inferView(Typer.scala:107)
[error] dotty.tools.dotc.typer.Typer.tryExtensionOrConversion(Typer.scala:3190)
[error] dotty.tools.dotc.typer.Typer.tryInsertImplicitOnQualifier$$anonfun$1(Typer.scala:3139)
[error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3035)
[error] dotty.tools.dotc.typer.Typer.tryInsertImplicitOnQualifier(Typer.scala:3143)
[error] dotty.tools.dotc.typer.Typer.tryImplicit$1(Typer.scala:3101)
[error] dotty.tools.dotc.typer.Typer.tryInsertApplyOrImplicit(Typer.scala:3114)
[error] dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:3906)
[error] dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3251)
[error] dotty.tools.dotc.typer.Typer.tryApply$1(Typer.scala:3097)
[error] dotty.tools.dotc.typer.Typer.tryInsertApplyOrImplicit$$anonfun$2(Typer.scala:3115)
[error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3035)
[error] dotty.tools.dotc.typer.Typer.tryInsertApplyOrImplicit(Typer.scala:3124)
[error] dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:3906)
[error] dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3251)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] dotty.tools.dotc.typer.Applications.typedTypeApply(Applications.scala:1087)
[error] dotty.tools.dotc.typer.Applications.typedTypeApply$(Applications.scala:317)
[error] dotty.tools.dotc.typer.Typer.typedTypeApply(Typer.scala:107)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2798)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:885)
[error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1035)
[error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[error] dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:107)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2779)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] dotty.tools.dotc.typer.Typer.typedMatch(Typer.scala:1498)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2793)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] dotty.tools.dotc.typer.Namer.typedAheadExpr$$anonfun$1(Namer.scala:1430)
[error] dotty.tools.dotc.typer.Namer.typedAhead(Namer.scala:1420)
[error] dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:1430)
[error] dotty.tools.dotc.typer.Namer.valOrDefDefSig(Namer.scala:1487)
[error] dotty.tools.dotc.typer.Namer.defDefSig(Namer.scala:1560)
[error] dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:756)
[error] dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:888)
[error] dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:780)
[error] dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:168)
[error] dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:188)
[error] dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:190)
[error] dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:370)
[error] dotty.tools.dotc.typer.Typer.retrieveSym(Typer.scala:2721)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2746)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2841)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2933)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2983)
[error] dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1044)
[error] dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1048)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2787)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.typedFunctionValue(Typer.scala:1427)
[error] dotty.tools.dotc.typer.Typer.makeContextualFunction(Typer.scala:2893)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2839)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:431)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:359)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:432)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:853)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:853)
[error] dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:544)
[error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:447)
[error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:735)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:852)
[error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1060)
[error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:317)
[error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:107)
[error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:907)
[error] dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$4(Applications.scala:986)
[error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3035)
[error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:997)
[error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1035)
[error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[error] dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:107)
[error] dotty.tools.dotc.typer.QuotesAndSplices.typedSplice(QuotesAndSplices.scala:118)
[error] dotty.tools.dotc.typer.QuotesAndSplices.typedSplice$(QuotesAndSplices.scala:35)
[error] dotty.tools.dotc.typer.Typer.typedSplice(Typer.scala:107)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2823)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2960)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2983)
[error] dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1044)
[error] dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1048)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2787)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:431)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:359)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:432)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:853)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:853)
[error] dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:544)
[error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:447)
[error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:735)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:852)
[error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1060)
[error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:317)
[error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:107)
[error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:907)
[error] dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$4(Applications.scala:986)
[error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3035)
[error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:997)
[error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1035)
[error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[error] dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:107)
[error] dotty.tools.dotc.typer.QuotesAndSplices.typedQuote(QuotesAndSplices.scala:69)
[error] dotty.tools.dotc.typer.QuotesAndSplices.typedQuote$(QuotesAndSplices.scala:35)
[error] dotty.tools.dotc.typer.Typer.typedQuote(Typer.scala:107)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2822)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] dotty.tools.dotc.typer.Typer.$anonfun$46(Typer.scala:2228)
[error] dotty.tools.dotc.typer.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:238)
[error] dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2228)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2756)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2841)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2933)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2983)
[error] dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2426)
[error] dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:2767)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2771)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2841)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2933)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2983)
[error] dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2553)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2812)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] dotty.tools.dotc.typer.TyperPhase.liftedTree1$1(TyperPhase.scala:56)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$1(TyperPhase.scala:62)
[error] dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:411)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck(TyperPhase.scala:63)
[error] dotty.tools.dotc.typer.TyperPhase.runOn$$anonfun$1(TyperPhase.scala:106)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.immutable.List.foreach(List.scala:333)
[error] dotty.tools.dotc.typer.TyperPhase.runOn(TyperPhase.scala:106)
[error] dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:261)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
[error] dotty.tools.dotc.Run.runPhases$1(Run.scala:272)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:280)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[error] dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:68)
[error] dotty.tools.dotc.Run.compileUnits(Run.scala:289)
[error] dotty.tools.dotc.Run.compileSources(Run.scala:222)
[error] dotty.tools.dotc.Run.compile(Run.scala:206)
[error] dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
[error] dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:192)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:247)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:182)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:163)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:163)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:210)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:528)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:528)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:175)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:173)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:459)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:414)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:501)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:401)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:167)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:528)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:482)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:420)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2357)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2314)
[error] sbt.internal.io.Retry$.apply(Retry.scala:46)
[error] sbt.internal.io.Retry$.apply(Retry.scala:28)
[error] sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:31)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2310)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] sbt.Execute.work(Execute.scala:291)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] java.base/java.lang.Thread.run(Thread.java:833)
[error]            
[error] java.lang.AssertionError: assertion failed: TS[2, 1, 0] attempted to take ownership of T which is already owned by committable TS[99, 2, 1, 0]
[error] 	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[error] 	at dotty.tools.dotc.core.TyperState.includeVar(TyperState.scala:240)
[error] 	at dotty.tools.dotc.core.TyperState.$anonfun$2$$anonfun$1(TyperState.scala:204)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.immutable.List.foreach(List.scala:333)
[error] 	at dotty.tools.dotc.core.TyperState.$anonfun$2(TyperState.scala:204)
[error] 	at scala.collection.immutable.List.forall(List.scala:386)
[error] 	at dotty.tools.dotc.core.TyperState.mergeConstraintWith(TyperState.scala:206)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArgs(ProtoTypes.scala:416)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.fold(ProtoTypes.scala:488)
[error] 	at dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:5921)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:5959)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.traverse(Implicits.scala:594)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$PolyProto.fold(ProtoTypes.scala:599)
[error] 	at dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:5921)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:5959)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.traverse(Implicits.scala:594)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$SelectionProto.fold(ProtoTypes.scala:214)
[error] 	at dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:5921)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:5959)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.traverse(Implicits.scala:594)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.apply(Types.scala:5958)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$ViewProto.fold(ProtoTypes.scala:545)
[error] 	at dotty.tools.dotc.core.Types$TypeAccumulator.foldOver(Types.scala:5921)
[error] 	at dotty.tools.dotc.core.Types$TypeTraverser.traverseChildren(Types.scala:5959)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.traverse(Implicits.scala:594)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo$collectParts$2$.apply(Implicits.scala:599)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo.recur$1(Implicits.scala:675)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo.computeIScope(Implicits.scala:686)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo.$anonfun$1(Implicits.scala:760)
[error] 	at dotty.tools.dotc.util.ReadOnlyMap.getOrElse(ReadOnlyMap.scala:23)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo.implicitScope(Implicits.scala:760)
[error] 	at dotty.tools.dotc.typer.ImplicitRunInfo.implicitScope$(Implicits.scala:543)
[error] 	at dotty.tools.dotc.Run.implicitScope(Run.scala:40)
[error] 	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.implicitScope(Implicits.scala:1454)
[error] 	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicit(Implicits.scala:1417)
[error] 	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicit(Implicits.scala:1426)
[error] 	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.bestImplicit(Implicits.scala:1451)
[error] 	at dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:991)
[error] 	at dotty.tools.dotc.typer.Implicits.inferImplicit$(Implicits.scala:771)
[error] 	at dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.Implicits.inferView(Implicits.scala:817)
[error] 	at dotty.tools.dotc.typer.Implicits.inferView$(Implicits.scala:771)
[error] 	at dotty.tools.dotc.typer.Typer.inferView(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.Typer.tryExtensionOrConversion(Typer.scala:3190)
[error] 	at dotty.tools.dotc.typer.Typer.tryInsertImplicitOnQualifier$$anonfun$1(Typer.scala:3139)
[error] 	at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3035)
[error] 	at dotty.tools.dotc.typer.Typer.tryInsertImplicitOnQualifier(Typer.scala:3143)
[error] 	at dotty.tools.dotc.typer.Typer.tryImplicit$1(Typer.scala:3101)
[error] 	at dotty.tools.dotc.typer.Typer.tryInsertApplyOrImplicit(Typer.scala:3114)
[error] 	at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:3906)
[error] 	at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3251)
[error] 	at dotty.tools.dotc.typer.Typer.tryApply$1(Typer.scala:3097)
[error] 	at dotty.tools.dotc.typer.Typer.tryInsertApplyOrImplicit$$anonfun$2(Typer.scala:3115)
[error] 	at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3035)
[error] 	at dotty.tools.dotc.typer.Typer.tryInsertApplyOrImplicit(Typer.scala:3124)
[error] 	at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:3906)
[error] 	at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3251)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 		at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] 	at dotty.tools.dotc.typer.Applications.typedTypeApply(Applications.scala:1087)
[error] 	at dotty.tools.dotc.typer.Applications.typedTypeApply$(Applications.scala:317)
[error] 	at dotty.tools.dotc.typer.Typer.typedTypeApply(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2798)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] 	at dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:885)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1035)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[error] 	at dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2779)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] 	at dotty.tools.dotc.typer.Typer.typedMatch(Typer.scala:1498)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2793)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] 	at dotty.tools.dotc.typer.Namer.typedAheadExpr$$anonfun$1(Namer.scala:1430)
[error] 	at dotty.tools.dotc.typer.Namer.typedAhead(Namer.scala:1420)
[error] 	at dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:1430)
[error] 	at dotty.tools.dotc.typer.Namer.valOrDefDefSig(Namer.scala:1487)
[error] 	at dotty.tools.dotc.typer.Namer.defDefSig(Namer.scala:1560)
[error] 	at dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:756)
[error] 	at dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:888)
[error] 	at dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:780)
[error] 	at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:168)
[error] 	at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:188)
[error] 	at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:190)
[error] 	at dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:370
[error] 	at dotty.tools.dotc.typer.Typer.retrieveSym(Typer.scala:2721)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2746)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2841)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2933)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2983)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1044)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1048)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2787)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.typedFunctionValue(Typer.scala:1427)
[error] 	at dotty.tools.dotc.typer.Typer.makeContextualFunction(Typer.scala:2893)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2839)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:431)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:359)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:432)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:853)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:853)
[error] 	at dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:544)
[error] 	at dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[error] 	at dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:447)
[error] 	at dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:735)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:852)
[error] 	at dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1060)
[error] 	at dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:317)
[error] 	at dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:907)
[error] 	at dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$4(Applications.scala:986)
[error] 	at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3035)
[error] 	at dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:997)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1035)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[error] 	at dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.QuotesAndSplices.typedSplice(QuotesAndSplices.scala:118)
[error] 	at dotty.tools.dotc.typer.QuotesAndSplices.typedSplice$(QuotesAndSplices.scala:35)
[error] 	at dotty.tools.dotc.typer.Typer.typedSplice(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2823)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2960)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2983)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1044)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1048)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2787)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:431)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:359)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:432)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:853)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:853)
[error] 	at dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:544)
[error] 	at dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:609)
[error] 	at dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:447)
[error] 	at dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:735)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:852)
[error] 	at dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1060)
[error] 	at dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:317)
[error] 	at dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:907)
[error] 	at dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$4(Applications.scala:986)
[error] 	at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3035)
[error] 	at dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:997)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1035)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:317)
[error] 	at dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.QuotesAndSplices.typedQuote(QuotesAndSplices.scala:69)
[error] 	at dotty.tools.dotc.typer.QuotesAndSplices.typedQuote$(QuotesAndSplices.scala:35)
[error] 	at dotty.tools.dotc.typer.Typer.typedQuote(Typer.scala:107)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2822)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] 	at dotty.tools.dotc.typer.Typer.$anonfun$46(Typer.scala:2228)
[error] 	at dotty.tools.dotc.typer.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:238)
[error] 	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2228)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2756)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2841)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2933)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2983)
[error] 	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2426)
[error] 	at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:2767)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2771)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2841)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2933)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2983)
[error] 	at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2553)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2812)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2842)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2907)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2911)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3027)
[error] 	at dotty.tools.dotc.typer.TyperPhase.liftedTree1$1(TyperPhase.scala:56)
[error] 	at dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$1(TyperPhase.scala:62)
[error] 	at dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:411)
[error] 	at dotty.tools.dotc.typer.TyperPhase.typeCheck(TyperPhase.scala:63)
[error] 	at dotty.tools.dotc.typer.TyperPhase.runOn$$anonfun$1(TyperPhase.scala:106)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.immutable.List.foreach(List.scala:333)
[error] 	at dotty.tools.dotc.typer.TyperPhase.runOn(TyperPhase.scala:106)
[error] 	at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:261)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
[error] 	at dotty.tools.dotc.Run.runPhases$1(Run.scala:272)
[error] 	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:280)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[error] 	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:68)
[error] 	at dotty.tools.dotc.Run.compileUnits(Run.scala:289)
[error] 	at dotty.tools.dotc.Run.compileSources(Run.scala:222)
[error] 	at dotty.tools.dotc.Run.compile(Run.scala:206)
[error] 	at dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
[error] 	at dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] 	at dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] 	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:192
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:247)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:182
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:163)
[error] 	at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:163)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:210)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:528)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:528)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:175)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:173)
[error] 	at sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:459)
[error] 	at sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] 	at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:414)
[error] 	at sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:501)
[error] 	at sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:401)
[error] 	at sbt.internal.inc.Incremental$.apply(Incremental.scala:167)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:528)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:482)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:420)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] 	at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2357)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2314)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:46)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:28)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] 	at sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:31)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2310)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] 	at java.base/java.lang.Thread.run(Thread.java:833)
[error] (Compile / compileIncremental) java.lang.AssertionError: assertion failed: TS[2, 1, 0] attempted to take ownership of T which is already owned by committable TS[99, 2, 1, 0]
[error] Total time: 3 s, completed 17 Dec 2021, 09:51:50

```
</details>

--- a/tests/neg-macros/i14123a.scala
+++ b/tests/neg-macros/i14123a.scala
@@ -1,0 +1,4 @@
+import scala.quoted._
+
+def f(foo: Any => Any)(using Quotes): Expr[Any] =
+  '{ println(${ foo[Int]('{???}); ??? }) } // error

--- a/tests/neg-macros/i14123b.scala
+++ b/tests/neg-macros/i14123b.scala
@@ -1,0 +1,23 @@
+package x
+
+import scala.quoted._
+
+object Impl {
+
+      sealed trait UpdateOp[+T]
+      case class Assignment[T](value:Expr[T]) extends UpdateOp[T]
+      case class Update(operation:Expr[Unit]) extends UpdateOp[Nothing]
+
+      def genRead[B:Type](newBuilder: Expr[B],
+                          readVal: (Expr[B]) => UpdateOp[B]
+                          )(using Quotes): Expr[B] =
+          '{
+                var x = $newBuilder
+                  ${readVal[B]('x) match { // error: method apply in trait Function1 does not take type parameters
+                    case Assignment(value) => '{ x = $value } // error
+                    case Update(operation) => operation // error
+                  }}
+                x
+          }
+
+}

